x-pack/test/profiling_api_integration/common/create_profiling_users/helpers/create_or_update_user.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/* eslint-disable no-console */
+
+import { difference, union } from 'lodash';
+import { Elasticsearch, Kibana } from '..';
+import { callKibana, isAxiosError } from './call_kibana';
+
+interface User {
+  username: string;
+  roles: string[];
+  full_name?: string;
+  email?: string;
+  enabled?: boolean;
+}
+
+export async function createOrUpdateUser({
+  elasticsearch,
+  kibana,
+  user,
+}: {
+  elasticsearch: Elasticsearch;
+  kibana: Kibana;
+  user: User;
+}) {
+  const existingUser = await getUser({
+    elasticsearch,
+    kibana,
+    username: user.username,
+  });
+  if (!existingUser) {
+    return createUser({ elasticsearch, kibana, newUser: user });
+  }
+
+  return updateUser({
+    elasticsearch,
+    kibana,
+    existingUser,
+    newUser: user,
+  });
+}
+
+async function createUser({
```

Comment:
__nit:__ I think you don't really want to rely on our internal APIs directly and should prefer security test services instead to make sure you don't need to update your tests any time we change our internal APIs: https://github.com/elastic/kibana/tree/main/test/common/services/security

Reply:
Thanks just replaced it by the security test. https://github.com/elastic/kibana/pull/159984/commits/5743646fbf7ad57aa9a67c9adec3e251aa21864b